### PR TITLE
feat: add prefix_scan API and CLI command

### PR DESCRIPTION
**Pull request by ben1009**
## Summary

Implements [RFC 006: Prefix Search](https://github.com/ben1009/toy-kv-engine/blob/main/rfcs/006-prefix-search.md).

Adds a native `prefix_scan(prefix)` API that returns all visible keys whose user key starts with a caller-provided byte prefix, in sorted key order. Internally maps the prefix to a lexicographic range `[prefix, next_prefix(prefix))` and delegates to the existing `scan` machinery — no new on-disk formats.

## Changes

- `prefix_upper_bound()` helper for computing exclusive upper bounds (handles `0xff` edge cases)
- `KvEngine::prefix_scan()` — convenience method over `scan`
- `Transaction::prefix_scan()` — snapshot reads with local-write merge, inherits serializable txn rejection
- `prefix <key>` CLI command
- 28 tests: unit (6), integration (8), MVCC (9), flush/compaction (6)

## Verification

- [x] `cargo fmt --check` passes
- [x] `cargo clippy -- -D warnings` passes
- [x] All 381 tests pass (28 new)

## Summary by CodeRabbit

* **New Features**
  * Prefix scanning to retrieve all key/value pairs matching a specified prefix.
  * New CLI command to run prefix-based queries.
  * Storage engine and transaction APIs support efficient prefix-based range scans.

* **Tests**
  * Comprehensive tests covering prefix bounds, edge cases, MVCC behavior, deletes, flush/compaction, and transaction interactions.

### CODE REVIEW

**Comment by ben1009** [comment]
CI Status: 16/17 checks passed. `ubuntu / nightly (unit)` rerunning — first run failed with exit code 100 (nightly toolchain flake, not a code issue). All stable-channel checks (ubuntu-latest, macOS, clippy, fmt, coverage, sanitizers) pass cleanly.

File: kv-engine/src/lsm_storage.rs [hunk 1]
```diff
@@ -180,6 +180,23 @@ pub enum CompactionFilter {
...
+    let mut upper = prefix.to_vec();
+    while let Some(last) = upper.last_mut() {
+        if *last != 0xff {
+            *last += 1;
+            return Some(upper);
+        }
+        upper.pop();
+    }
+    None
+}
```
[the remark this answers is not in the document]
Comment (ben1009) [reply kv-engine/src/lsm_storage.rs:200]: Fixed in 1e6829d. Now scans from the end to find the last non-0xff byte first, then allocates only the needed slice. Thanks for the suggestion!

**Comment by ben1009** [comment]
@coderabbitai review

**Comment by ben1009** [comment]
@gemini-code-assist review
